### DISCUSSION TITLE
Downgrade from 1.22.2 to 1.22.0

### DISCUSSION THREAD

**Discussion by zuhair68** [Koenkk/zigbee2mqtt#10577] [General]
I want to downgrade from 1.22.2 to 1.22.0
Will this script work?

```
sudo systemctl stop zigbee2mqtt
cd /opt/zigbee2mqtt
cp -R data data-backup
git checkout HEAD -- npm-shrinkwrap.json
git clone https://github.com/Koenkk/zigbee2mqtt
git checkout tags/1.22.0
npm ci
cp -R data-backup/* data
rm -rf data-backup
sudo systemctl start zigbee2mqtt
```

**Comment by daserra23** [comment]
Also interested in this, I can't find downgrade instructions anywhere.

**Comment by jeroenvdd** [comment]
if you want to downgrade also motivate why, always interesting for Koen to know if there are breaking issues

**Reply by daserra23** [reply]
I created another post with a problem I have with an Aqara switch, was hoping to try a downgrade to see if it is a hardware/firmware fault or something in the new release.

**Comment by oberschwabe** [comment]
it works
with small adjustments in the i managed to downgrade with your script from 1.25.2 to 1.25.0
thanks a lot

**Comment by jarrah31** [comment]
Based on this guide for swapping to dev branch, I just ran the following:
https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html#linux

```
# Stop Zigbee2MQTT and go to directory
sudo systemctl stop zigbee2mqtt
cd /opt/zigbee2mqtt

# Backup configuration
cp -R data data-backup

# Update
git fetch
git checkout tags/1.25.2 # Change to 'master' to switch back to the release version
git pull
npm ci

# Restore configuration
cp -R data-backup/* data
rm -rf data-backup

# Start Zigbee2MQTT
sudo systemctl start zigbee2mqtt
```

I've no idea why all the Linux upgrade instructions ask you to restore data-backup into data. The data folder stays intact during an upgrade with no modifications, and I've managed fine without doing that for many upgrades now.
